drivers/led_strip/ws2812_spi.c

```diff
@@ -151,6 +186,12 @@ static int ws2812_spi_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	if (cfg->bits_per_symbol < 3 || cfg->bits_per_symbol > 8) {
```

Comment:
This should be possible to check compile time with `BUILD_ASSERT()` in `WS2812_SPI_DEVICE()`.